p2p/profile_service.go

```diff
@@ -0,0 +1,218 @@
+package p2p
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/helpers"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/qri-io/qri/config"
+	"github.com/qri-io/qri/event"
+	"github.com/qri-io/qri/repo"
+	"github.com/qri-io/qri/repo/profile"
+)
+
+// ProfileProtocolID is the protocol id for the profile exchange service
+const ProfileProtocolID = "/qri/profile/0.1.0"
+
+// ProfileTimeout is the length of time we will wait for a response in a profile
+// exchange
+const ProfileTimeout = time.Minute * 2
+
+// ProfileService manages the profile exchange. This exchange should happen
+// whenever a node connects to a new peer
+type ProfileService struct {
+	Host    host.Host
+	Repo    repo.Repo
+	pub     event.Publisher
+	peersMu *sync.Mutex
```

Comment:
whenever you declare a mutex lock, add a comment about the fields it's governing: `// lock for peers map`